server/auth_middleware.go

```
@@ -0,0 +1,20 @@
+package server
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/A9u/function_junction/db"
+	"github.com/mongodb/mongo-go-driver/bson/primitive"
+)
+
+func AuthMiddleware(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	ctx := r.Context()
+	user_id := r.Header.Get("Authorization")
```

We should never ignore errors in Go.